NitroxPatcher/Patches/Dynamic/Vehicle_OnKill_Patch.cs

```diff
@@ -12,8 +15,13 @@ public class Vehicle_OnKill_Patch : NitroxPatch, IDynamicPatch
         public static readonly MethodInfo TARGET_METHOD = TARGET_CLASS.GetMethod("OnKill", BindingFlags.NonPublic | BindingFlags.Instance);
 
         public static void Prefix(Vehicle __instance)
-        {
-            NitroxServiceLocator.LocateService<Vehicles>().BroadcastDestroyedVehicle(__instance);
+        {            
+            NitroxId id = NitroxEntity.GetId(__instance.gameObject);
+            SimulationOwnership simulationOwnership = NitroxServiceLocator.LocateService<SimulationOwnership>();
+            if (simulationOwnership.HasAnyLockType(id) || simulationOwnership.OtherPlayerHasAnyLock(id))
```

Comment:
Why even check for a lock? in both cases you stop simulating.

Reply:
Could happen that the client does not have notice of a lock. Removing the simulation would then raise an error.